lib/puppet/provider/keycloak_realm/kcadm.rb

```diff
@@ -226,6 +260,23 @@ def create
         raise Puppet::Error, "kcadm update realms/#{resource[:name]}/default-optional-client-scopes/#{scope_id}: #{e.message}"
       end
     end
+    if resource[:roles]
+      roles = @property_hash[:roles] ||= get_realm_roles(resource[:name])
```

Comment:
Per other comment , just make this :

```
role = get_realm_roles(resource[:name])
```

Also move this logic under the other `if resource[:roles]` so only doing the `get_realm_roles` query once.